lib/Doctrine/ORM/Mapping/ClassMetadata.php

```diff
@@ -4,25 +4,3410 @@
+     * when their properties change. Such entity classes must implement
+     * the <tt>NotifyPropertyChanged</tt> interface.
+     */
+    public const CHANGETRACKING_NOTIFY = 3;
+
+    /**
+     * Specifies that an association is to be fetched when it is first accessed.
+     */
+    public const FETCH_LAZY = 2;
+
+    /**
+     * Specifies that an association is to be fetched when the owner of the
+     * association is fetched.
+     */
+    public const FETCH_EAGER = 3;
+
+    /**
+     * Specifies that an association is to be fetched lazy (on first access) and that
+     * commands such as Collection#count, Collection#slice are issued directly against
+     * the database if the collection is not yet initialized.
+     */
+    public const FETCH_EXTRA_LAZY = 4;
+
+    /**
+     * Identifies a one-to-one association.
+     */
+    public const ONE_TO_ONE = 1;
+
+    /**
+     * Identifies a many-to-one association.
+     */
+    public const MANY_TO_ONE = 2;
+
+    /**
+     * Identifies a one-to-many association.
+     */
+    public const ONE_TO_MANY = 4;
+
+    /**
+     * Identifies a many-to-many association.
+     */
+    public const MANY_TO_MANY = 8;
+
     /**
-     * Repeating the ClassMetadataInfo constructor to infer correctly the template with PHPStan
+     * Combined bitmask for to-one (single-valued) associations.
+     */
+    public const TO_ONE = 3;
+
+    /**
+     * Combined bitmask for to-many (collection-valued) associations.
+     */
+    public const TO_MANY = 12;
+
+    /**
+     * ReadOnly cache can do reads, inserts and deletes, cannot perform updates or employ any locks,
+     */
+    public const CACHE_USAGE_READ_ONLY = 1;
+
+    /**
+     * Nonstrict Read Write Cache doesn’t employ any locks but can do inserts, update and deletes.
+     */
+    public const CACHE_USAGE_NONSTRICT_READ_WRITE = 2;
+
+    /**
+     * Read Write Attempts to lock the entity before update/delete.
+     */
+    public const CACHE_USAGE_READ_WRITE = 3;
+
+    /**
+     * The value of this column is never generated by the database.
+     */
+    public const GENERATED_NEVER = 0;
+
+    /**
+     * The value of this column is generated by the database on INSERT, but not on UPDATE.
+     */
+    public const GENERATED_INSERT = 1;
+
+    /**
+     * The value of this column is generated by the database on both INSERT and UDPATE statements.
+     */
+    public const GENERATED_ALWAYS = 2;
+
+    /**
+     * READ-ONLY: The name of the entity class.
      *
-     * @see https://github.com/doctrine/orm/issues/8709
+     * @var string
+     * @psalm-var class-string<T>
+     */
+    public $name;
+
+    /**
+     * READ-ONLY: The namespace the entity class is contained in.
      *
-     * @param string $entityName The name of the entity class the new instance is used for.
-     * @psalm-param class-string<T> $entityName
+     * @var string
+     * @todo Not really needed. Usage could be localized.
```

Comment:
I'm not sure what "Usage could be localized" means.

Reply:
:thinking: me neither